js/stdlib/ts/ALGO.ts

```diff
@@ -2305,49 +2315,54 @@ const balancesOfM = async (acc: Account, tokens: Array<Token|null>): Promise<Arr
   return Promise.all(tokenBalances);
 };
 
+const balanceOfMDhead = 'balanceOfM';
+
 const balanceOfM = async (acc: Account, token: Token | null): Promise<BigNumber | false> => {
-  const dhead = 'balanceOfM';
   const addr = extractAddr(acc);
-  if (await nodeCanRead()) {
-    const client = await getAlgodClient();
-    if (token == null) {
+  if (token == null) {
+    if (await nodeCanRead()) {
+      const client = await getAlgodClient();
       // Node algo balance
       const query = client.accountInformation(addr).exclude('all') as unknown as ApiCall<AccountInfo>;
-      const accountInfo = await doQuery_(dhead, query);
+      const accountInfo = await doQuery_(balanceOfMDhead, query);
       return bigNumberify(accountInfo['amount']);
     } else {
-      // Node token balance
-      const query = client.accountAssetInformation(addr, bigNumberToNumber(token)) as unknown as ApiCall<AccountAssetInfo>;
-      const accountAssetInfoM = await doQueryM_(dhead, query);
-      if ('val' in accountAssetInfoM) {
-        return bigNumberify(accountAssetInfoM['val']['asset-holding']?.['amount'] ?? 0);
-      } else {
-        return false;
-      }
-    }
-  } else {
-    const indexer = await getIndexer();
-    if (token == null) {
+      const indexer = await getIndexer();
       // Indexer algo balance
       const query = indexer.lookupAccountByID(addr).exclude('all') as unknown as ApiCall<IndexerAccountInfoRes>;
-      const accountInfoM = await doQueryM_(dhead, query);
+      const accountInfoM = await doQueryM_(balanceOfMDhead, query);
       if ('val' in accountInfoM) {
         return bigNumberify(accountInfoM['val']['account']['amount']);
       } else {
         return false;
       }
+    }
+  } else {
+    return balanceOfMNonNet(addr, token);
+  }
+}
+
+const balanceOfMNonNet = async (addr:string, token:Token) => {
+  // TODO - I think these can and should be merged.  I think the only meaningful difference is that one case returns 0 where the other returns false, which is probably a bug due to drift.  But I'm not sure which side is correct.
```

Comment:
I want to remove the if statement and just use one side.  My guess is that if there is no `asset-holding` field then the return should be false, IE the second case, but I'm not 100% certain.  I could try simplifying to each side and see what tests break both ways, but thought I'd ask first.

Reply:
Oh, I see I missed some of your comments from earlier, including one that talks about this.

Reply:
Yea